src/FirstSteps/RPG/Adventures/Mine.cs

```diff
@@ -19,9 +19,11 @@ public void Enter(Hero hero)
 
             enter += 1;
 
-            var coins = enter % 2 == 0 ? 3 : 1;
+            var coins = enter % 2 == 0 ? 3 : enter % 5 == 0 ?  +2 : enter == 13 ? 0 : 1;
```

Comment:
It would be best if you did not use the '?:' notation many times in one line because it is super unclear for someone that looks the first time at the code. Shortening is not always a good choice.
Maybe try to follow that description:
1. If your visit is the thirteenth visit return zero as soon as possible. 
2. Calculate the number of coins based on the condition of an even visit. 
3. If your visit is divisible by five add two more coins to the result.

Reply:
You can take a look one more time at this commend and take into account the order of actions in points 1, 2, and 3.